### ISSUE TITLE
Image axis cannot be customized

### ISSUE THREAD

**Issue by robin-cls** [holoviz/geoviews#552]
Hi,

I tried to follow the customization guidelines https://holoviews.org/user_guide/Customizing_Plots.html#axis-ticks and apply them on an image but I couldn't get the same results.

I produced a minimal example, for which I plot two backends bokeh and matplotlib. I expected the two plots to have the following behavior :
- With the xticks options, only the three mentionned values 105, 125, 145 should be displayed on the X axis. However, both bokeh and matplolib display all the ticks
- With the xformatter option, I expected the longitudes and latitudes to be displayed in bokeh as matplotlib does (aka. the longitude 120 is written 120°E)

A little more context : when producting maps of sea level, we wish to be able to display a given subset of longitudes and latitudes over the axis, and sometimes without subticks. Also, it is a plus to be able to relabel the longitude with the desired unit (°S and °E) to relate to the plot readers' habits

```
import geoviews as gv
import numpy as np

gv.extension("bokeh", "matplotlib")

image = gv.Image(
    (
        np.arange(100, 150),
        np.arange(10, 40),
        np.ones((30, 50))
    )
).opts(
    xticks=[105, 125, 145],
    xformatter="%f°E",
    yformatter="%f°N",
)

gv.output(image, backend="bokeh")
gv.output(image, backend="matplotlib")
```

![image](https://user-images.githubusercontent.com/74916839/150531503-59d56939-02ef-40f2-8b00-ac328aa6921b.png)

![image](https://user-images.githubusercontent.com/74916839/150531471-428a300d-9372-492f-b29e-839d84158896.png)

holoviews : 1.14.6
geoviews : 1.9.2

**Comment by jbednar** [comment]
I don't think this issue is specific to geoviews, so I'll use a HoloViews example here. Moreover, I think what you are seeing is due to how options are handled per backend, not to do with Images or axes in particular. Options in HoloViews are always stored _per_ _backend_, because they aren't typically the same across backends. These particular options happen to be supported in both Matplotlib and Bokeh, and in that case, you can declare that HoloViews should apply them to both backends:

```python
import holoviews as hv, numpy as np
hv.extension("bokeh", "matplotlib")
aopts=dict(
    xticks=[105, 125, 145],
    xformatter="%f°E",
    yformatter="%f°N",
)

image = hv.Image((np.arange(100, 150), np.arange(10, 40), np.ones((30, 50))))
image.opts(backend="bokeh", **aopts)
image.opts(backend="matplotlib", **aopts)

hv.output(image, backend="bokeh")
hv.output(image, backend="matplotlib")
```

![image](https://user-images.githubusercontent.com/1695496/150606836-ff318949-b004-4a27-a9a9-b1bac1fc7658.png)

If you don't specify the backend when you provide an option, it will default to the currently active backend, which starts out with whatever backend is listed first in `hv.extension()` and then I believe changes whenever you do `hv.output`. If you want to support switching back and forth, you should make sure you're always declaring which backend your options apply to, or else apply them in clearly separated phases where you first supply Matplotlib options as a group, then switch to Bokeh and supply those options, or vice versa.
